datafusion/physical-expr/src/aggregate/median.rs

```diff
@@ -151,7 +151,12 @@ impl Accumulator for MedianAccumulator {
                 // ignore null values
                 .filter(|v| !v.is_null())
                 .cloned(),
-        )?;
+        )
+        .map_err(|_| {
+            DataFusionError::Execution(
+                "median requires at least one non-null value".into(),
+            )
+        })?;
```

Comment:
Does this have the possibility of incorrectly overwriting other errors that might have occurred in `iter_to_array`?

Reply:
I think so.

What about explicitly checking for all nulls with something like:

```rust
if  !self.all_values
                .iter()
                .any(|v| !v.is_null()) {
.. raise error here
}
```